modules/dcache/src/main/java/org/dcache/pool/statistics/IoStatisticsChannel.java

```diff
@@ -0,0 +1,168 @@
+package org.dcache.pool.statistics;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.dcache.pool.repository.RepositoryChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.SyncFailedException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * This class decorates any RepositoryChannel and adds the function of collecting data for the IoStatistics
+ */
+public class IoStatisticsChannel implements RepositoryChannel {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(IoStatisticsChannel.class);
+
+    /**
+     * Inner channel to which all operations are delegated.
+     */
+    @VisibleForTesting
+    private RepositoryChannel _channel;
+
+    private final IoStatistics _statistics = new IoStatistics();
+
+    public IoStatisticsChannel(RepositoryChannel channel){ _channel = channel; }
+
+    /**
+     * Returns the object most central of this decorator
+     * @return object with collected and evaluated statistics data
+     */
+    public IoStatistics getStatistics() {
+        return _statistics;
+    }
+
+    @Override
+    public int write(ByteBuffer buffer, long position) throws IOException {
+        long startTime = System.currentTimeMillis();
```

Comment:
The IO request can be quite small and with SSD ~200 microseconds. Please use System.nanoTime() to calculate time delta. See http://download.java.net/java/jdk9/docs/api/java/lang/System.html#nanoTime-- for details.